test/run_test.py

```diff
@@ -487,7 +487,12 @@ def run_test(
         os.close(log_fd)
 
     command = (launcher_cmd or []) + executable + argv
-    should_retry = "--subprocess" not in command and not RERUN_DISABLED_TESTS
+    should_retry = (
+        "--subprocess" not in command
+        and not RERUN_DISABLED_TESTS
+        and not is_cpp_test
```

Comment:
Just curious, does rerun work with C++ tests, maybe not resuming from the last failed tests, but retrying the whole test file N times?  It has been a while since I looked at them.